test/e2e/operator/deployment_api.go

```diff
@@ -235,6 +235,56 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 				return validate.DriverDeployment(client, k8sver, d.Namespace, deployment)
 			}, "1m", "20s").ShouldNot(HaveOccurred(), "driver validation failure after restarting")
 		})
+
+		It("shall recover from conflicts", func() {
+			deployment := getDeployment("test-recover-from-conflicts")
+			sec := &corev1.Secret{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Secret",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      deployment.GetHyphenedName() + "-registry-secrets",
+					Namespace: d.Namespace,
+				},
+				Type: corev1.SecretTypeTLS,
+				Data: map[string][]byte{
+					"ca.crt":  []byte("fake ca"),
+					"tls.key": []byte("fake key"),
+					"tls.crt": []byte("fake crt"),
+				},
+			}
+			deleteSecret := func(name string) {
+				Eventually(func() error {
+					err := f.ClientSet.CoreV1().Secrets(d.Namespace).Delete(context.Background(), name, metav1.DeleteOptions{})
+					deploy.LogError(err, "Delete secret error: %v, will retry...", err)
+					if errors.IsNotFound(err) {
+						return nil
+					}
+					return err
+				}, "3m", "1s").Should(BeNil(), "delete secret %q", name)
```

Comment:
Instead of `BeNil`, always check errors with `ShouldNot+HaveOccurred` - it produces better output when the check fails.

Reply:
Changed to `HaveOccurred()`